sdk/src/main/java/com/google/cloud/dataflow/sdk/io/BigQueryIO.java

```diff
@@ -523,6 +549,12 @@ public void validate(PInput input) {
           } else if (query != null && flattenResults == null) {
             throw new IllegalStateException("Invalid BigQuery read operation. Specifies a"
                 + " query without a result flattening preference");
+          } else if (table != null && useLegacySql != null) {
```

Comment:
This follows the pattern of flattenResults.

However, I think we can move the validation of useLegacySql to usingStandardSql(). For example, throws if users do BigQuery.Read.from(table).usingStandardSql().

Also, looks to me flattenResults is not configurable, and can be removed in a separate PR.